sources/Algo/include/Algo.h

```diff
@@ -192,13 +192,24 @@ class GeneratorDefinitionAlgorithm : public NodeAlgorithm {
   /**
    * @brief Perform algorithm
    *
-   * Add the generators of the nodes and deducing the model to use
+   * Add the generators of the nodes and deducing the model to use. If the node is a generator with a deduced
```

Comment:
Please rephrase: for example "validity of the generator is checked whenever it makes sense"